db/range_tombstone_fragmenter.h

```diff
@@ -0,0 +1,95 @@
+//  Copyright (c) 2018-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <vector>
+#include <list>
+#include <string>
+#include <memory>
+
+#include "rocksdb/status.h"
+#include "db/dbformat.h"
+#include "db/pinned_iterators_manager.h"
+#include "table/internal_iterator.h"
+
+namespace rocksdb {
+
+// FragmentedRangeTombstoneIterator converts an InternalIterator of a range-del
+// meta block into an iterator over non-overlapping tombstone fragments. The
+// tombstone fragmentation process should be more efficient than the range
+// tombstone collapsing algorithm in RangeDelAggregator because this leverages
+// the internal key ordering already provided by the input iterator. If there
```

Comment:
the range tombstone meta-blocks were not sorted in older versions, I believe. That was a change Nikhil made.

Reply:
Thanks for pointing this out. I've removed that assumption while trying to make the sorted case fast. I think there's still some slowdown though, so I'll run the benchmarks again and compare the results.

Reply:
Yeah, it slowed down by about 3x. Here's the updated `readrandom` result (using the same commands as in the PR description):
```
readrandom   :      11.147 micros/op 89707 ops/sec;    6.9 MB/s (63103 of 100000 found)
```

Reply:
Fortunately, I think the effect is minimal in #4493; I ran some cursory benchmarks on that PR after rebasing and they don't seem to have changed.